integration/live_update_after_crash_rebuild_test.go

```diff
@@ -0,0 +1,76 @@
+//+build integration
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLiveUpdateAfterCrashRebuild(t *testing.T) {
+	f := newK8sFixture(t, "live_update_after_crash_rebuild")
+	defer f.TearDown()
+	f.SetRestrictedCredentials()
+
+	f.TiltWatch()
+
+	ctx, cancel := context.WithTimeout(f.ctx, time.Minute)
+	defer cancel()
+	fmt.Println("> Waiting for pods from initial build")
+	oneUpPods := f.WaitForAllPodsReady(ctx, "app=live-update-after-crash-rebuild")
+
+	ctx, cancel = context.WithTimeout(f.ctx, time.Minute)
+	defer cancel()
+	f.CurlUntil(ctx, "http://localhost:31234", "🍄 One-Up! 🍄")
+
+	// Live update
+	f.ReplaceContents("compile.sh", "One-Up", "Two-Up")
+
+	ctx, cancel = context.WithTimeout(f.ctx, time.Minute)
+	defer cancel()
+	fmt.Println("> Perform a live update")
+	f.CurlUntil(ctx, "http://localhost:31234", "🍄 Two-Up! 🍄")
+
+	// Check that the pods were changed in place, and that we didn't create new ones
+	twoUpPods := f.WaitForAllPodsReady(ctx, "app=live-update-after-crash-rebuild")
+	require.Equal(t, oneUpPods, twoUpPods)
+
+	if len(twoUpPods) != 1 {
+		t.Fatalf("Expected one pod, actual: %v", twoUpPods)
+	}
+
+	// Delete the pod and make sure it got replaced with something that prints the
+	// same thing (crash rebuild).
+	fmt.Println("> Kill pod, wait for crash rebuild")
+	f.runCommandSilently("kubectl", "delete", "pod", twoUpPods[0], namespaceFlag)
+
+	ctx, cancel = context.WithTimeout(f.ctx, time.Minute)
+	defer cancel()
+	f.CurlUntil(ctx, "http://localhost:31234", "🍄 Two-Up! 🍄")
+
+	// Unfortunately "WaitForAllPodsReady" isn't that accurate and can pull in terminating pods
```

Comment:
This surprises me. If it pulls in a "terminating" pod, wouldn't that be not ready, and thus either way, it won't return until the terminating pod is gone and the new pod is ready?

Or is that "terminating" with a lower-case t, as in "maybe we killed it but it's still 'Running' for an instant, so we might return from `WaitForAllPodsReady` because the old pod is still running and the new pod doesn't exist yet"? (at which point we should probably say "the pod we just deleted" or something rather than using a term that is also the name of a state returned by kubectl)

In which case, it seems like we should have the sleep before the `CurlUntil` above, because that state would also mean we're getting a Curl back from the old pod, but the goal is to get it from the new pod?

This is maybe all nitty since this is probably copypasta'd from another test that works well enough, and maybe really more of a separate PR issue - just trying to 1) figure out what this is supposed to be doing and 2) figure out if we can make the code more directly match the intent

Reply:
yeah i don't get what's going on here in any detail, but here's my report: https://windmill-eng.slack.com/archives/C9SA451CJ/p1572559413240100

Reply:
it's the second case you mentioned: we killed it but it's still running for an instant. i'll poke around at reordering the sleep/the other stuff and update the comment to make more sense.